diff --git a/scilab/configure b/scilab/configure
index 32433e8..779b8c2 100755
--- a/scilab/configure
+++ b/scilab/configure
 # Gets compilation and library flags
 
 
+####FOR LIBARCHIVE#############################################################################################
+
+LIBARCHIVE_REQUIRED_VERSION=3.0.0
+
+pkg_failed=no
+{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for LIBARCHIVE" >&5
+$as_echo_n "checking for LIBARCHIVE... " >&6; }
+
+if test -n "$PKG_CONFIG"; then
+    if test -n "$LIBARCHIVE_CFLAGS"; then
+        pkg_cv_LIBARCHIVE_CFLAGS="$LIBARCHIVE_CFLAGS"
+    else
+        if test -n "$PKG_CONFIG" && \
+    { { $as_echo "$as_me:${as_lineno-$LINENO}: \$PKG_CONFIG --exists --print-errors \"libarchive >= \$LIBARCHIVE_REQUIRED_VERSION\""; } >&5
+  ($PKG_CONFIG --exists --print-errors "libarchive >= $LIBARCHIVE_REQUIRED_VERSION") 2>&5
+  ac_status=$?
+  $as_echo "$as_me:${as_lineno-$LINENO}: \$? = $ac_status" >&5
+  test $ac_status = 0; }; then
+  pkg_cv_ARCHIVE_CFLAGS=`$PKG_CONFIG --cflags "libarchive >= $LIBARCHIVE_REQUIRED_VERSION" 2>/dev/null`
+else
+  pkg_failed=yes
+fi
+    fi
+else
+	pkg_failed=untried
+fi
+
+if test -n "$PKG_CONFIG"; then
+    if test -n "$LIBARCHIVE_LIBS"; then
+        pkg_cv_LIBARCHIVE_LIBS="$LIBARCHIVE_LIBS"
+    else
+        if test -n "$PKG_CONFIG" && \
+    { { $as_echo "$as_me:${as_lineno-$LINENO}: \$PKG_CONFIG --exists --print-errors \"libarchive >= \$LIBARCHIVE_REQUIRED_VERSION\""; } >&5
+  ($PKG_CONFIG --exists --print-errors "libarchive >= $LIBARCHIVE_REQUIRED_VERSION") 2>&5
+  ac_status=$?
+  $as_echo "$as_me:${as_lineno-$LINENO}: \$? = $ac_status" >&5
+  test $ac_status = 0; }; then
+  pkg_cv_LIBARCHIVE_LIBS=`$PKG_CONFIG --libs "libarchive >= $LIBARCHIVE_REQUIRED_VERSION" 2>/dev/null`
+else
+  pkg_failed=yes
+fi
+    fi
+else
+	pkg_failed=untried
+fi
+
+
+
+if test $pkg_failed = yes; then
+
+if $PKG_CONFIG --atleast-pkgconfig-version 0.20; then
+        _pkg_short_errors_supported=yes
+else
+        _pkg_short_errors_supported=no
+fi
+        if test $_pkg_short_errors_supported = yes; then
+	        LIBARCHIVE_PKG_ERRORS=`$PKG_CONFIG --short-errors --errors-to-stdout --print-errors "libarchive >= $LIBARCHIVE_REQUIRED_VERSION"`
+        else
+	        LIBARCHIVE_PKG_ERRORS=`$PKG_CONFIG --errors-to-stdout --print-errors "libarchive >= $LIBARCHIVE_REQUIRED_VERSION"`
+        fi
+	# Put the nasty error message in config.log where it belongs
+	echo "$LIBARCHIVE_PKG_ERRORS" >&5
+
+	as_fn_error $? "Package requirements (libarchive >= $LIBARCHIVE_REQUIRED_VERSION) were not met:
+
+$LIBARCHIVE_PKG_ERRORS
+
+Consider adjusting the PKG_CONFIG_PATH environment variable if you
+installed software in a non-standard prefix.
+
+Alternatively, you may set the environment variables CURL_CFLAGS
+and CURL_LIBS to avoid the need to call pkg-config.
+See the pkg-config man page for more details.
+" "$LINENO" 5
+elif test $pkg_failed = untried; then
+	{ { $as_echo "$as_me:${as_lineno-$LINENO}: error: in \`$ac_pwd':" >&5
+$as_echo "$as_me: error: in \`$ac_pwd':" >&2;}
+as_fn_error $? "The pkg-config script could not be found or is too old.  Make sure it
+is in your PATH or set the PKG_CONFIG environment variable to the full
+path to pkg-config.
+
+Alternatively, you may set the environment variables CURL_CFLAGS
+and CURL_LIBS to avoid the need to call pkg-config.
+See the pkg-config man page for more details.
+
+To get pkg-config, see <http://pkg-config.freedesktop.org/>.
+See \`config.log' for more details" "$LINENO" 5; }
+else
+	LIBARCHIVE_CFLAGS=$pkg_cv_LIBARCHIVE_CFLAGS
+	LIBARCHIVE_LIBS=$pkg_cv_LIBARCHIVE_LIBS
+        { $as_echo "$as_me:${as_lineno-$LINENO}: result: yes" >&5
+$as_echo "yes" >&6; }
+	:
+fi
+
+LIBS="$LIBARCHIVE_LIBS $LIBS"
+CFLAGS="$LIBARCHIVE_CFLAGS $CFLAGS"
+
+LIBS="$saved_LIBS"
+CFLAGS="$saved_cflags"
+
+
+
+
+
+
+cat >>confdefs.h <<_ACEOF
+#define LIBARCHIVE_CFLAGS "$_CFLAGS"
+_ACEOF
+
+
+cat >>confdefs.h <<_ACEOF
+#define LIBARCHIVE_LIBS "$LIBARCHIVE_LIBS"
+_ACEOF
+
+
+cat >>confdefs.h <<_ACEOF
+#define LIBARCHIVE_VERSION "$LIBARCHIVE_VERSION"
+_ACEOF
+
+
+################################################################################################################
 #################
 ## Tcl/Tk library
 #################
@@ -46284,6 +46421,10 @@ echo "  CURL_CFLAGS ........ = $CURL_CFLAGS"
 echo "  CURL_LIBS .......... = $CURL_LIBS"
 echo "  CURL_VERSION ....... = $CURL_VERSION"
 
+echo ""
+echo "Libarchive Configuration:"
+echo " LIBARCHIVE_CFLAGS.....= $LIBARCHIVE_CFLAGS"
+echo " LIBARCHIVE_LIBS.......= $LIBARCHIVE_LIBS"
 
 echo ""
 echo "SWIG Configuration:"
